go/inst/instance_dao.go

```diff
@@ -744,6 +761,10 @@ func ReadTopologyInstanceBufferable(instanceKey *InstanceKey, bufferWrites bool,
 	}
 
 	{
+		// wait for routines to finish before reading cluster attributes. This seems to be needed, as,
+		// otherwise, detecting whether a host is a replication group secondary to set up cluster attributes from its primary
+		// does not work. ToDo: Is there a better way to sync this?
+		waitGroup.Wait()
```

Comment:
this seems wrong, why do you need to wait here?

Reply:
Without this, it would sometimes think that a group member was not actually a group member, because the `ReplicationGroupName` attribute was not populated yet. Hence I added the wait.

Reply:
This will hurt parallelism. I see the dependency problem. Let me see if moving `ReadInstanceClusterAttributes` itself can  break dependencies.

Reply:
this shouldn't be in a wait group since it has to be applied in sequential order, same as how `show slave status` query is read.

I would put this after `show slave status` code:
```go
instance.ReplicationIOThreadState = ReplicationThreadStateNoThread
instance.ReplicationSQLThreadState = ReplicationThreadStateNoThread
err = sqlutils.QueryRowsMap(db, "show slave status", func(m sqlutils.RowMap) error {
...
if err != nil {
	goto Cleanup
}
if instance.IsOracleMySQL() && !instance.IsSmallerMajorVersionByString("5.7") {
	err = PopulateGroupReplicationInformation(instance, db)
        if err != nil {
	        goto Cleanup
        }
}
```